### ISSUE TITLE
Rule define-macros-order don't allow require statements

### ISSUE THREAD

**Issue by Zehir** [vuejs/eslint-plugin-vue#1946]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.22.0
- **eslint-plugin-vue version:** 9.3.0
- **Node version:** v16.15.1
- **Operating System:** Windows, Github codespace

Using "@antfu/eslint-config": "^0.26.1"

**Please show your full configuration:**

```json5
{
  "extends": "@antfu"
}
```

**What did you do?**	

```vue
<script setup lang="ts">
import Color = require('color')
import type { GatewayCredentials } from '~/interfaces/deconz'

const props = defineProps<{
  modelValue: GatewayCredentials
}>()

</script>
```

**What did you expect to happen?**
I expect that I should not have an warning on the defineProps of the type `vue/define-macros-order`

**What actually happened?**

```
defineProps should be the first statement in `<script setup>` (after any potential import statements or type definitions).eslint[vue/define-macros-order](https://eslint.vuejs.org/rules/define-macros-order.html)
```

If I let VS Code fix the problem (with the button `Fix this `vue/define-macros-order problem` , I get this result:
```vue
<script setup lang="ts">
const props = defineProps<{
  modelValue: GatewayCredentials
}>()
import Color = require('color')
import type { GatewayCredentials } from '~/interfaces/deconz'

</script>
```
There is also a format on save routine, I get a new result on each save

**Repository to reproduce this issue**

You can find my repo to reproduce [here](https://github.com/Zehir/deconz-ruler/commit/003834ea94ed4b08588463be47ece66c07f41542)
The file is located at `src/components/gateway-badge.vue`

**Comment by Zehir** [comment]
I just check, it's not working, maybe it's not the correct way to import the `color` module, let me check

**Comment by Zehir** [comment]
Ok it's working with `import Color from 'color'` I don't know why it's was not working first
